docs/reference/modules/discovery.asciidoc

```diff
@@ -1,30 +1,811 @@
 [[modules-discovery]]
-== Discovery
+== Discovery and cluster formation
 
-The discovery module is responsible for discovering nodes within a
-cluster, as well as electing a master node.
+The discovery and cluster formation module is responsible for discovering
+nodes, electing a master, forming a cluster, and publishing the cluster state
+each time it changes. It is integrated with other modules, for example, all
+communication between nodes is done using the <<modules-transport,transport>>
+module.
 
-Note, Elasticsearch is a peer to peer based system, nodes communicate
-with one another directly if operations are delegated / broadcast. All
-the main APIs (index, delete, search) do not communicate with the master
-node. The responsibility of the master node is to maintain the global
-cluster state, and act if nodes join or leave the cluster by reassigning
-shards. Each time a cluster state is changed, the state is made known to
-the other nodes in the cluster (the manner depends on the actual
-discovery implementation).
+It is separated into several sections, which are explained below:
```

Comment:
I think "it" in this sentence is referring to the documentation and the list that follows is a list of links to subsequent sections. If that's right, my advice would be to (1) Split this one very long page into separate pages and (2) Remove this bulleted list and turn it into brief paragraphs on the (short) introductory page. If you agree, I can help with these changes.

Reply:
I made a few changes here as you suggested, including splitting the page up now that we're happy with the doc structure. The latest version 00a0145a2366edf105724871e8d7d4323598c79e has all of these changes.